File: pset1/Mario/mario.c
#include <stdio.h>
#include <cs50.h>

void printSpaces(int count);
void printHashes(int count);
void print(char character, int count);
void finishLine();

int main()
{
    int x = -1;
    while (x > 23 || x < 0)
    {
        printf("Height: ");
        x = get_int();
    }

    for (int i = 0; i < x; i++)
    {
        int spaces = x - i - 1;
        int hashes = x - spaces + 1;

        printSpaces(spaces);
        printHashes(hashes);
        finishLine();
    }
}

// prints space N times
void printSpaces(int count)
{
    print(' ', count);
}

// prints hash N times
void printHashes(int count)
{
    print('#', count);
}

// Prints character N times
void print(char character, int count)
{
    for (int j = 0; j < count; j++)
    {
        printf("%c", character);
    }
}

void finishLine()
{
    printf("\n");
}

File: pset7/mario/mario.py
from cs50 import get_int

x = -1
# get value from the user
while x > 23 or x < 0:
    x = get_int("Height: ")

# print the stairs
for i in range(x):
    spaces = x - i - 1
    hashes = x - spaces + 1

    print(" " * spaces, end="")
    print("#" * hashes, end="")
    print("")